modules/transactionpool/standard.go

```diff
@@ -0,0 +1,43 @@
+package transactionpool
+
+import (
+	"errors"
+
+	"github.com/NebulousLabs/Sia/consensus"
+	"github.com/NebulousLabs/Sia/encoding"
+)
+
+const (
+	TransactionSizeLimit = 64 * 1024
+)
+
+// standard implements the rules outlined in Standard.md, and will return an
+// error if any of the rules are violated.
+func standard(t consensus.Transaction) (err error) {
+	// Check that the size of the transaction does not exceed the standard
+	// established in Standard.md
+	encodedTxn := encoding.Marshal(t)
+	if len(encodedTxn) > TransactionSizeLimit {
+		err = errors.New("transaction is too large")
+		return
+	}
+
+	// Check that transactions with storage proofs follow the storage proof
+	// rules.
+	if len(t.StorageProofs) != 0 {
+		if len(t.Inputs) > 1 ||
+			len(t.MinerFees) > 1 ||
+			len(t.Outputs) > 1 ||
+			len(t.FileContracts) != 0 ||
+			len(t.ArbitraryData) != 0 {
+			err = errors.New("transaction has storage proofs but does not follow the storage proof rules")
```

Comment:
it's only partially generic, you get several bits of information:

- if the transaction is too large
- if the transaction violates storage proof rules
- if the transaction violates arbitrary data rules